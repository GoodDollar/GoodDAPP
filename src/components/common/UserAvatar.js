// @flow
import React from 'react'
import { StyleSheet, View } from 'react-native'
import { normalize } from 'react-native-elements'
import CreateAvatar from 'exif-react-avatar-edit'
import { getScreenHeight, getScreenWidth, isPortrait } from '../../lib/utils/Orientation'

import Avatar from './Avatar'
import Section from './Section'

export type AvatarProps = {
  profile: {
    avatar: string,
    fullName?: string
  },
  onChange?: any => mixed,
  onClose?: any => mixed,
  originalSize?: boolean,
  editable?: boolean,
  children?: React.Node
}

/**
 * Touchable Users Avatar based on Avatar component
 * @param {AvatarProps} props
 * @param {Object} props.profile
 * @param {string} props.profile.avatar
 * @param {string} props.profile.fullName
 * @param {any => mixed} props.onChange
 * @param {any => mixed} props.onClose
 * @param {boolean} props.originalSize
 * @param {boolean} props.editable
 * @param {React.Node} props.children
 * @returns {React.Node}
 */
const UserAvatar = (props: AvatarProps) => {
  const { profile, editable, onChange, onClose, originalSize = false, children } = props
  let cropSize = isPortrait() ? getScreenWidth() - 70 : getScreenHeight() - 70
  if (cropSize > 320) {
    cropSize = 320
  }

  return editable ? (
    <View style={styles.innerAvatar}>
      <View style={styles.cropContainer}>
        <CreateAvatar
          onCrop={onChange}
          onClose={onClose}
          mobileScaleSpeed={0.01}
          width={cropSize}
          height={cropSize}
          lineWidth={2}
          minCropRadius={15}
          shadingOpacity={0.8}
          src={profile.avatar ? profile.avatar : undefined}
        />
      </View>
    </View>
  ) : (
    <View style={styles.avatar}>
      <View style={styles.innerAvatar}>
        <Avatar size={originalSize ? cropSize : 120} {...props} source={profile.avatar}>
          {children}
        </Avatar>
        <Section.Title>{profile.fullName}</Section.Title>
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  avatar: {
    marginTop: normalize(50),
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row'
  },
  innerAvatar: {
    flexDirection: 'column'
  },
  fullNameContainer: {
    padding: 10
  },
  fullName: {
    textAlign: 'left'
  },
  cropContainer: {
    marginTop: normalize(5),
    flex: 1,
    justifyContent: 'center',
    flexDirection: 'row'
  }
})

export default UserAvatar
