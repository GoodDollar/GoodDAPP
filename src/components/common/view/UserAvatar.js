// @flow
import React from 'react'
import { View } from 'react-native'
import { getDesignRelativeWidth } from '../../../lib/utils/sizes'

import { withStyles } from '../../../lib/styles'

import Avatar from './Avatar'

const AVATAR_DESIGN_WIDTH = 136

export type AvatarProps = {
  profile: {
    avatar: string,
    fullName?: string,
  },
  onChange?: any => mixed,
  onClose?: any => mixed,
  originalSize?: boolean,
  editable?: boolean,
  children?: React.Node,
}

/**
 * Touchable Users Avatar based on Avatar component
 * @param {AvatarProps} props
 * @param {Object} props.profile
 * @param {string} props.profile.avatar
 * @param {string} props.profile.fullName
 * @param {any => mixed} props.onChange
 * @param {any => mixed} props.onClose
 * @param {boolean} props.editable
 * @param {React.Node} props.children
 * @returns {React.Node}
 */
const UserAvatar = ({ profile, children, styles, containerStyle, size, ...rest }: AvatarProps) => (
  <View style={styles.avatar}>
    <View style={[styles.innerAvatar, containerStyle]}>
      <Avatar size={size} {...rest} source={profile.avatar}>
        {children}
      </Avatar>
    </View>
  </View>
)

UserAvatar.defaultProps = {
  size: getDesignRelativeWidth(AVATAR_DESIGN_WIDTH),
}

const getStylesFromProps = ({ theme, size }) => ({
  avatar: {
    flexDirection: 'row',
    justifyContent: 'center',
    width: size,
  },
  innerAvatar: {
    alignItems: 'center',
    flex: 1,
    flexDirection: 'column',
  },
  fullNameContainer: {
    marginTop: theme.sizes.default,
  },
  fullName: {
    textAlign: 'left',
  },
  cropContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: theme.paddings.mainContainerPadding,
  },
})

export default withStyles(getStylesFromProps)(UserAvatar)
