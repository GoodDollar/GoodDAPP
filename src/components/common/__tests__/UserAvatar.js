import React from 'react'
import UserAvatar from '../UserAvatar'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('UserAvatar', () => {
  const profile = {
    fullName: 'John Doe',
    email: 'a****a@aaa.com',
    mobile: '*********444'
  }

  it('renders without errors', () => {
    const tree = renderer.create(<UserAvatar profile={profile} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<UserAvatar profile={profile} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
