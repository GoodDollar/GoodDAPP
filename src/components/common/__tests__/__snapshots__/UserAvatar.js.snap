// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxDirection": "normal",
      "WebkitBoxFlex": 1,
      "WebkitBoxOrient": "horizontal",
      "WebkitBoxPack": "center",
      "WebkitFlexBasis": "0%",
      "WebkitFlexDirection": "row",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "WebkitJustifyContent": "center",
      "flexBasis": "0%",
      "flexDirection": "row",
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "marginTop": "50px",
      "msFlexDirection": "row",
      "msFlexNegative": 1,
      "msFlexPack": "center",
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexDirection": "column",
        "flexDirection": "column",
        "msFlexDirection": "column",
      }
    }
  >
    <div
      aria-disabled={true}
      className="css-view-1dbjc4n"
      disabled={true}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "MozTransitionProperty": "opacity",
          "MozUserSelect": "none",
          "WebkitTransitionDuration": "0.15s",
          "WebkitTransitionProperty": "opacity",
          "WebkitUserSelect": "none",
          "backgroundColor": "rgba(0,0,0,0.00)",
          "msUserSelect": "none",
          "transitionDuration": "0.15s",
          "transitionProperty": "opacity",
          "userSelect": "none",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,1.00)",
            "borderBottomColor": "rgba(112,112,112,1.00)",
            "borderBottomLeftRadius": "60px",
            "borderBottomRightRadius": "60px",
            "borderBottomWidth": "1px",
            "borderLeftColor": "rgba(112,112,112,1.00)",
            "borderLeftWidth": "1px",
            "borderRightColor": "rgba(112,112,112,1.00)",
            "borderRightWidth": "1px",
            "borderTopColor": "rgba(112,112,112,1.00)",
            "borderTopLeftRadius": "60px",
            "borderTopRightRadius": "60px",
            "borderTopWidth": "1px",
            "height": "120px",
            "width": "120px",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitFlexBasis": "auto",
              "borderBottomLeftRadius": "60px",
              "borderBottomRightRadius": "60px",
              "borderTopLeftRadius": "60px",
              "borderTopRightRadius": "60px",
              "flexBasis": "auto",
              "height": "120px",
              "msFlexPreferredSize": "auto",
              "overflowX": "hidden",
              "overflowY": "hidden",
              "width": "120px",
              "zIndex": 0,
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "backgroundPosition": "center",
                "backgroundRepeat": "no-repeat",
                "backgroundSize": "cover",
                "bottom": "0px",
                "height": "100%",
                "left": "0px",
                "position": "absolute",
                "right": "0px",
                "top": "0px",
                "width": "100%",
                "zIndex": -1,
              }
            }
          />
        </div>
      </div>
    </div>
    <div
      className="css-text-901oao"
      dir="auto"
      style={
        Object {
          "color": "rgba(85,85,85,1.00)",
          "fontSize": "18px",
          "textAlign": "center",
        }
      }
    >
      John Doe
    </div>
  </div>
</div>
`;
